Generation not correct?

### Tell us the extension you're using

Client

### I tried this:

This is a follow-on from #1185 

Again I'm unsure whether the problem is in the OpenAPI document itself, or whether it is in the generation.

Here's the OpenAPI document (updated version than what was in #1185 - supposedly the "fixed" something)
[docling.json](https://github.com/user-attachments/files/20839295/docling.json)

### This happened:

When I ran the generation, this is what I got:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,62] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,67] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,151] ';' expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,164] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,165] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,168] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,188] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,324] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,366] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,507] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[24,666] ';' expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[25,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[26,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[34,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[36,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[41,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[43,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[45,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[47,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[48,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[55,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[57,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[62,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[64,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[66,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[68,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[69,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[70,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[73,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[75,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[76,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[83,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[85,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[90,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[92,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[94,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[96,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[97,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[98,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[101,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[103,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[104,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[110,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[112,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[114,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[115,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[116,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[118,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[119,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[120,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[133,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[135,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[137,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[139,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[142,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[150,12] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[154,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[163,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[165,5] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,66] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,71] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,155] ';' expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,168] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,169] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,172] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,192] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,328] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,370] <identifier> expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,511] illegal start of expression
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[171,670] ';' expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[173,17] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[175,17] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[182,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[184,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[189,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[191,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[193,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[195,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[196,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[203,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[205,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[210,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[212,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[214,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[216,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[217,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[218,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[220,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[221,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[228,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[230,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[235,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[237,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[239,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[241,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[242,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[243,16] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[245,13] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[246,9] class, interface, enum, or record expected
[ERROR] /Users/edeandre/workspaces/quarkus/quarkus-docling/runtime/target/generated-sources/open-api/io/quarkiverse/docling/runtime/client/model/ConversionRequest.java:[252,16] class, interface, enum, or record expected
```

## Full build output log

[build-output.txt](https://github.com/user-attachments/files/20839374/build-output.txt)

### I expected this:

_No response_

### Is there a workaround?

_No response_

### How can we try to reproduce the issue?

1. Clone https://github.com/quarkiverse/quarkus-docling
2. `gh pr checkout 31`
3. Download [docling.json](https://github.com/user-attachments/files/20839349/docling.json) and put it in `runtime/src/main/openapi`
4. `./mvnw -f runtime clean package -DskipTests`

### Anything else?

Here is the complete file it generated:

```java
package io.quarkiverse.docling.runtime.client.model;


import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonTypeName;
import com.fasterxml.jackson.annotation.JsonValue;
import io.quarkiverse.docling.runtime.client.model.ConvertDocumentFileSourcesRequest;
import io.quarkiverse.docling.runtime.client.model.ConvertDocumentHttpSourcesRequest;
import io.quarkiverse.docling.runtime.client.model.ConvertDocumentsOptions1;
import io.quarkiverse.docling.runtime.client.model.FileSource1;
import io.quarkiverse.docling.runtime.client.model.HttpSource1;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import com.fasterxml.jackson.annotation.JsonProperty;

@com.fasterxml.jackson.annotation.JsonIgnoreProperties(ignoreUnknown = true)
public class ConversionRequest  {

    private ConvertDocumentsOptions1 options = {from_formats=[docx, pptx, html, image, pdf, asciidoc, md, csv, xlsx, xml_uspto, xml_jats, json_docling], to_formats=[md], image_export_mode=embedded, do_ocr=true, force_ocr=false, ocr_engine=easyocr, pdf_backend=dlparse_v4, table_mode=accurate, pipeline=standard, page_range=[1, 9223372036854776000], document_timeout=604800, abort_on_error=false, return_as_file=false, do_table_structure=true, include_images=true, images_scale=2, md_page_break_placeholder=, do_code_enrichment=false, do_formula_enrichment=false, do_picture_classification=false, do_picture_description=false, picture_description_area_threshold=0.05};
    private List<FileSource1> fileSources = new ArrayList<>();
    private List<HttpSource1> httpSources = new ArrayList<>();

    /**
    * Get options
    * @return options
    **/
    @JsonProperty("options")
    @com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)
    public ConvertDocumentsOptions1 getOptions() {
        return options;
    }

    /**
     * Set options
     **/
    public void setOptions(ConvertDocumentsOptions1 options) {
        this.options = options;
    }

    public ConversionRequest options(ConvertDocumentsOptions1 options) {
        this.options = options;
        return this;
    }

    /**
    * Get fileSources
    * @return fileSources
    **/
    @JsonProperty("file_sources")
    public List<FileSource1> getFileSources() {
        return fileSources;
    }

    /**
     * Set fileSources
     **/
    public void setFileSources(List<FileSource1> fileSources) {
        this.fileSources = fileSources;
    }

    public ConversionRequest fileSources(List<FileSource1> fileSources) {
        this.fileSources = fileSources;
        return this;
    }
    public ConversionRequest addFileSourcesItem(FileSource1 fileSourcesItem) {
        if (this.fileSources == null){
            fileSources = new ArrayList<>();
        }
        this.fileSources.add(fileSourcesItem);
        return this;
    }

    /**
    * Get httpSources
    * @return httpSources
    **/
    @JsonProperty("http_sources")
    public List<HttpSource1> getHttpSources() {
        return httpSources;
    }

    /**
     * Set httpSources
     **/
    public void setHttpSources(List<HttpSource1> httpSources) {
        this.httpSources = httpSources;
    }

    public ConversionRequest httpSources(List<HttpSource1> httpSources) {
        this.httpSources = httpSources;
        return this;
    }
    public ConversionRequest addHttpSourcesItem(HttpSource1 httpSourcesItem) {
        if (this.httpSources == null){
            httpSources = new ArrayList<>();
        }
        this.httpSources.add(httpSourcesItem);
        return this;
    }

    /**
     * Create a string representation of this pojo.
     **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class ConversionRequest {\n");

        sb.append("    options: ").append(toIndentedString(options)).append("\n");
        sb.append("    fileSources: ").append(toIndentedString(fileSources)).append("\n");
        sb.append("    httpSources: ").append(toIndentedString(httpSources)).append("\n");
        
        sb.append("}");
        return sb.toString();
    }

    /**
     * Compares this object to the specified object. The result is
     * {@code true} if and only if the argument is not
     * {@code null} and is a {@code ConversionRequest} object that
     * contains the same values as this object.
     *
     * @param   obj   the object to compare with.
     * @return  {@code true} if the objects are the same;
     *          {@code false} otherwise.
     **/
    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;

        ConversionRequest model = (ConversionRequest) obj;

        return java.util.Objects.equals(options, model.options) &&
        java.util.Objects.equals(fileSources, model.fileSources) &&
        java.util.Objects.equals(httpSources, model.httpSources);
    }

    /**
     * Returns a hash code for a {@code ConversionRequest}.
     *
     * @return a hash code value for a {@code ConversionRequest}.
     **/
    @Override
    public int hashCode() {
        return java.util.Objects.hash(options,
        fileSources,
        httpSources);
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private static String toIndentedString(Object o) {
        if (o == null) {
            return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }

    @com.fasterxml.jackson.annotation.JsonIgnoreProperties(ignoreUnknown = true)
    public static class ConversionRequestQueryParam  {

        @jakarta.ws.rs.QueryParam("options")
        private ConvertDocumentsOptions1 options = {from_formats=[docx, pptx, html, image, pdf, asciidoc, md, csv, xlsx, xml_uspto, xml_jats, json_docling], to_formats=[md], image_export_mode=embedded, do_ocr=true, force_ocr=false, ocr_engine=easyocr, pdf_backend=dlparse_v4, table_mode=accurate, pipeline=standard, page_range=[1, 9223372036854776000], document_timeout=604800, abort_on_error=false, return_as_file=false, do_table_structure=true, include_images=true, images_scale=2, md_page_break_placeholder=, do_code_enrichment=false, do_formula_enrichment=false, do_picture_classification=false, do_picture_description=false, picture_description_area_threshold=0.05};
        @jakarta.ws.rs.QueryParam("fileSources")
        private List<FileSource1> fileSources = new ArrayList<>();
        @jakarta.ws.rs.QueryParam("httpSources")
        private List<HttpSource1> httpSources = new ArrayList<>();

        /**
        * Get options
        * @return options
        **/
        @com.fasterxml.jackson.annotation.JsonProperty("options")
        public ConvertDocumentsOptions1 getOptions() {
            return options;
        }

        /**
         * Set options
         **/
        public void setOptions(ConvertDocumentsOptions1 options) {
            this.options = options;
        }

        public ConversionRequestQueryParam options(ConvertDocumentsOptions1 options) {
            this.options = options;
            return this;
        }

        /**
        * Get fileSources
        * @return fileSources
        **/
        @com.fasterxml.jackson.annotation.JsonProperty("file_sources")
        public List<FileSource1> getFileSources() {
            return fileSources;
        }

        /**
         * Set fileSources
         **/
        public void setFileSources(List<FileSource1> fileSources) {
            this.fileSources = fileSources;
        }

        public ConversionRequestQueryParam fileSources(List<FileSource1> fileSources) {
            this.fileSources = fileSources;
            return this;
        }
        public ConversionRequestQueryParam addFileSourcesItem(FileSource1 fileSourcesItem) {
            this.fileSources.add(fileSourcesItem);
            return this;
        }

        /**
        * Get httpSources
        * @return httpSources
        **/
        @com.fasterxml.jackson.annotation.JsonProperty("http_sources")
        public List<HttpSource1> getHttpSources() {
            return httpSources;
        }

        /**
         * Set httpSources
         **/
        public void setHttpSources(List<HttpSource1> httpSources) {
            this.httpSources = httpSources;
        }

        public ConversionRequestQueryParam httpSources(List<HttpSource1> httpSources) {
            this.httpSources = httpSources;
            return this;
        }
        public ConversionRequestQueryParam addHttpSourcesItem(HttpSource1 httpSourcesItem) {
            this.httpSources.add(httpSourcesItem);
            return this;
        }

        /**
         * Create a string representation of this pojo.
         **/
        @Override
        public String toString() {
            StringBuilder sb = new StringBuilder();
            sb.append("class ConversionRequestQueryParam {\n");

            sb.append("    options: ").append(toIndentedString(options)).append("\n");
            sb.append("    fileSources: ").append(toIndentedString(fileSources)).append("\n");
            sb.append("    httpSources: ").append(toIndentedString(httpSources)).append("\n");
            sb.append("}");
            return sb.toString();
        }

        /**
         * Convert the given object to string with each line indented by 4 spaces
         * (except the first line).
         */
        private static String toIndentedString(Object o) {
            if (o == null) {
                return "null";
            }
            return o.toString().replace("\n", "\n    ");
        }
    }}
```

### Output of `uname -a` or `ver`

Darwin edeandrea-m4max 24.5.0 Darwin Kernel Version 24.5.0: Tue Apr 22 19:53:27 PDT 2025; root:xnu-11417.121.6~2/RELEASE_ARM64_T6041 arm64

### Output of `java -version`

openjdk version "21.0.6" 2025-01-21 LTS OpenJDK Runtime Environment Temurin-21.0.6+7 (build 21.0.6+7-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.6+7 (build 21.0.6+7-LTS, mixed mode, sharing)

### Quarkus OpenApi version or git rev

2.11.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937) Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.9.9/3477a4f1 Java version: 21.0.6, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/21.0.6-tem Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "15.5", arch: "aarch64", family: "mac"

### Additional information

_No response_

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.name: Bug Report 🐞

--- a/client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java
+++ b/client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java
@@ -24,6 +24,7 @@ import org.openapitools.codegen.languages.JavaClientCodegen;
 import org.openapitools.codegen.model.ModelMap;
 import org.openapitools.codegen.model.OperationMap;
 import org.openapitools.codegen.model.OperationsMap;
+import org.openapitools.codegen.utils.ModelUtils;
 import org.openapitools.codegen.utils.ProcessUtils;
 import org.openapitools.codegen.utils.URLPathUtils;
 import org.slf4j.Logger;
@@ -226,6 +227,16 @@ public class QuarkusJavaClientCodegen extends JavaClientCodegen {
             }
         }
         return super.fromProperty(name, p, required, schemaIsFromAdditionalProperties);
+    }
+
+    @Override
+    public String toDefaultValue(CodegenProperty property, Schema schema) {
+        Schema referenceSchema = ModelUtils.getReferencedSchema(this.openAPI, schema);
+        if (ModelUtils.isObjectSchema(referenceSchema)
+                && referenceSchema.getDefault() != null) {
+            return null;
+        }
+        return super.toDefaultValue(property, schema);
     }
 
     private void warnIfDuplicated(CodegenModel m) {

--- a/client/deployment/src/main/resources/openapi/issue-1185-object-default.json
+++ b/client/deployment/src/main/resources/openapi/issue-1185-object-default.json
@@ -1,0 +1,83 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "API with object default values",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/v1/convert": {
+      "post": {
+        "summary": "Convert Document",
+        "operationId": "convertDocument",
+        "tags": ["Conversion"],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/ConversionRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "ConversionRequest": {
+        "type": "object",
+        "title": "ConversionRequest",
+        "properties": {
+          "options": {
+            "$ref": "#/components/schemas/ConvertDocumentsOptions"
+          }
+        }
+      },
+      "ConvertDocumentsOptions": {
+        "type": "object",
+        "title": "ConvertDocumentsOptions",
+        "description": "Options for document conversion.",
+        "properties": {
+          "from_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": ["docx", "pptx", "html", "image", "pdf", "asciidoc", "md"],
+            "description": "Input format(s) to convert from."
+          },
+          "to_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": ["md"],
+            "description": "Output format(s) to convert to."
+          },
+          "ocr_engine": {
+            "type": "object",
+            "title": "Ocr Engine Options",
+            "description": "OCR engine options.",
+            "default": {
+              "ocr_engine": "easyocr",
+              "lang": "en"
+            }
+          }
+        }
+      }
+    }
+  }
+}

--- a/client/deployment/src/main/resources/openapi/issue-1216-schema-default.json
+++ b/client/deployment/src/main/resources/openapi/issue-1216-schema-default.json
@@ -1,0 +1,166 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "API with object schema default value via $ref",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/v1/convert": {
+      "post": {
+        "summary": "Convert Document",
+        "operationId": "convertDocument",
+        "tags": ["Conversion"],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/ConversionRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "ConversionRequest": {
+        "type": "object",
+        "title": "ConversionRequest",
+        "properties": {
+          "options": {
+            "$ref": "#/components/schemas/ConvertDocumentsOptions"
+          }
+        }
+      },
+      "ConvertDocumentsOptions": {
+        "type": "object",
+        "title": "ConvertDocumentsOptions",
+        "description": "Options for document conversion.",
+        "default": {
+          "from_formats": ["docx", "pptx", "html", "image", "pdf", "asciidoc", "md"],
+          "to_formats": ["md"],
+          "image_export_mode": "embedded",
+          "do_ocr": true,
+          "force_ocr": false,
+          "ocr_engine": "easyocr",
+          "pdf_backend": "dlparse_v4",
+          "table_mode": "accurate",
+          "pipeline": "standard",
+          "document_timeout": 604800,
+          "abort_on_error": false,
+          "return_as_file": false,
+          "do_table_structure": true,
+          "include_images": true,
+          "images_scale": 2,
+          "do_code_enrichment": false,
+          "do_formula_enrichment": false,
+          "do_picture_classification": false,
+          "do_picture_description": false,
+          "picture_description_area_threshold": 0.05
+        },
+        "properties": {
+          "from_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "description": "Input format(s) to convert from."
+          },
+          "to_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "description": "Output format(s) to convert to."
+          },
+          "image_export_mode": {
+            "type": "string",
+            "description": "Image export mode."
+          },
+          "do_ocr": {
+            "type": "boolean",
+            "description": "If enabled, the bitmap content will be processed using OCR."
+          },
+          "force_ocr": {
+            "type": "boolean",
+            "description": "If enabled, replace existing text with OCR-generated text."
+          },
+          "ocr_engine": {
+            "type": "string",
+            "description": "The OCR engine to use."
+          },
+          "pdf_backend": {
+            "type": "string",
+            "description": "The PDF backend to use."
+          },
+          "table_mode": {
+            "type": "string",
+            "description": "Mode to use for table structure."
+          },
+          "pipeline": {
+            "type": "string",
+            "description": "Choose the pipeline to process PDF or image files."
+          },
+          "document_timeout": {
+            "type": "integer",
+            "description": "The timeout for processing each document, in seconds."
+          },
+          "abort_on_error": {
+            "type": "boolean",
+            "description": "Abort on error if enabled."
+          },
+          "return_as_file": {
+            "type": "boolean",
+            "description": "Return the output as a zip file."
+          },
+          "do_table_structure": {
+            "type": "boolean",
+            "description": "If enabled, the table structure will be extracted."
+          },
+          "include_images": {
+            "type": "boolean",
+            "description": "If enabled, images will be extracted from the document."
+          },
+          "images_scale": {
+            "type": "number",
+            "description": "Scale factor for images."
+          },
+          "do_code_enrichment": {
+            "type": "boolean",
+            "description": "If enabled, perform OCR code enrichment."
+          },
+          "do_formula_enrichment": {
+            "type": "boolean",
+            "description": "If enabled, perform formula OCR, return LaTeX code."
+          },
+          "do_picture_classification": {
+            "type": "boolean",
+            "description": "If enabled, classify pictures in documents."
+          },
+          "do_picture_description": {
+            "type": "boolean",
+            "description": "If enabled, describe pictures in documents."
+          },
+          "picture_description_area_threshold": {
+            "type": "number",
+            "description": "Minimum percentage of the area for a picture to be processed."
+          }
+        }
+      }
+    }
+  }
+}

--- a/client/deployment/src/test/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapperTest.java
+++ b/client/deployment/src/test/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapperTest.java
@@ -900,6 +900,62 @@ public class OpenApiClientGeneratorWrapperTest {
     }
 
     @Test
+    void verifyObjectDefaultValuesDoNotGenerateInvalidJava() throws Exception {
+        List<File> generatedFiles = createGeneratorWrapper("issue-1185-object-default.json")
+                .generate("org.issue1185objectdefault")
+                .stream()
+                .filter(file -> file.getPath().endsWith("ConvertDocumentsOptions.java")).toList();
+
+        assertThat(generatedFiles).isNotEmpty();
+        for (File file : generatedFiles) {
+            try {
+                CompilationUnit cu = StaticJavaParser.parse(file);
+                List<FieldDeclaration> fields = cu.findAll(FieldDeclaration.class);
+                for (FieldDeclaration field : fields) {
+                    for (VariableDeclarator var : field.getVariables()) {
+                        Expression init = var.getInitializer().orElse(null);
+                        if (init != null) {
+                            String initStr = init.toString();
+                            assertFalse(initStr.startsWith("{") && !initStr.startsWith("{\""),
+                                    "Field " + var.getNameAsString() + " has invalid Java default value: " + initStr);
+                        }
+                    }
+                }
+            } catch (FileNotFoundException e) {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
+    }
+
+    @Test
+    void verifySchemaDefaultValuesDoNotGenerateInvalidJava() throws Exception {
+        List<File> generatedFiles = createGeneratorWrapper("issue-1216-schema-default.json")
+                .generate("org.issue1216schemadefault")
+                .stream()
+                .filter(file -> file.getPath().endsWith("ConversionRequest.java")).toList();
+
+        assertThat(generatedFiles).isNotEmpty();
+        for (File file : generatedFiles) {
+            try {
+                CompilationUnit cu = StaticJavaParser.parse(file);
+                List<FieldDeclaration> fields = cu.findAll(FieldDeclaration.class);
+                for (FieldDeclaration field : fields) {
+                    for (VariableDeclarator var : field.getVariables()) {
+                        Expression init = var.getInitializer().orElse(null);
+                        if (init != null) {
+                            String initStr = init.toString();
+                            assertFalse(initStr.startsWith("{") && !initStr.startsWith("{\""),
+                                    "Field " + var.getNameAsString() + " has invalid Java default value: " + initStr);
+                        }
+                    }
+                }
+            } catch (FileNotFoundException e) {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
+    }
+
+    @Test
     void verifySingleMethodForAllMediaTypesGeneration() throws java.net.URISyntaxException, FileNotFoundException {
         OpenApiClientGeneratorWrapper generatorWrapper = createGeneratorWrapper("issue-1428.yaml")
                 .withMethodPerMediaType(false);

--- a/server/deployment/src/main/java/io/quarkiverse/openapi/server/generator/deployment/codegen/openapitools/QuarkusJavaServerCodegen.java
+++ b/server/deployment/src/main/java/io/quarkiverse/openapi/server/generator/deployment/codegen/openapitools/QuarkusJavaServerCodegen.java
@@ -3,7 +3,11 @@ package io.quarkiverse.openapi.server.generator.deployment.codegen.openapitools;
 import static io.quarkiverse.openapi.server.generator.deployment.codegen.openapitools.QuarkusJavaServerCodegenConfigurator.USE_BEAN_VALIDATION;
 import static io.quarkiverse.openapi.server.generator.deployment.codegen.openapitools.QuarkusJavaServerCodegenConfigurator.USE_REST_RESPONSE;
 
+import org.openapitools.codegen.CodegenProperty;
 import org.openapitools.codegen.languages.JavaClientCodegen;
+import org.openapitools.codegen.utils.ModelUtils;
+
+import io.swagger.v3.oas.models.media.Schema;
 
 public class QuarkusJavaServerCodegen extends JavaClientCodegen {
 
@@ -58,6 +62,28 @@ public class QuarkusJavaServerCodegen extends JavaClientCodegen {
 
         this.modelTemplateFiles.clear();
         this.modelTemplateFiles.put("model.qute", ".java");
+    }
+
+    @Override
+    public CodegenProperty fromProperty(String name, Schema p, boolean required, boolean schemaIsFromAdditionalProperties) {
+        if (p != null && p.getType() != null) {
+            if ("object".equals(p.getType())
+                    && p.getDefault() != null
+                    && p.getAdditionalProperties() == null
+                    && p.getItems() == null) {
+                p.setAdditionalProperties(true);
+            }
+        }
+        return super.fromProperty(name, p, required, schemaIsFromAdditionalProperties);
+    }
+
+    @Override
+    public String toDefaultValue(CodegenProperty property, Schema schema) {
+        Schema referenceSchema = ModelUtils.getReferencedSchema(this.openAPI, schema);
+        if (ModelUtils.isObjectSchema(referenceSchema) && referenceSchema.getDefault() != null) {
+            return null;
+        }
+        return super.toDefaultValue(property, schema);
     }
 
 }

--- a/server/deployment/src/test/java/io/quarkiverse/openapi/server/generator/deployment/codegen/openapitools/OpenAPIToolsServerCodegenTest.java
+++ b/server/deployment/src/test/java/io/quarkiverse/openapi/server/generator/deployment/codegen/openapitools/OpenAPIToolsServerCodegenTest.java
@@ -226,6 +226,54 @@ class OpenAPIToolsServerCodegenTest {
                 .contains("AnAnnotation");
     }
 
+    @Test
+    @DisplayName("Object default values should not generate invalid Java")
+    void should_not_generate_invalid_java_for_object_default_values() throws IOException {
+        Path path = findOpenAPIPath("issue-1216-object-default.json");
+
+        OpenAPIToolsGenerator openAPIToolsGenerator = new OpenAPIToolsGenerator(
+                new QuarkusJavaServerCodegenConfigurator()
+                        .withInputBaseDir(path.toString())
+                        .withOutputDir(Files.createTempDirectory("").toString())
+                        .withBasePackage("org.issue1216objectdefault"));
+
+        List<File> files = openAPIToolsGenerator.generate();
+
+        List<File> modelFiles = files.stream()
+                .filter(file -> file.getName().endsWith("ConvertDocumentsOptions.java"))
+                .toList();
+
+        assertThat(modelFiles).isNotEmpty();
+        for (File file : modelFiles) {
+            String content = Files.readString(file.toPath());
+            assertThat(content).doesNotContain("= {");
+        }
+    }
+
+    @Test
+    @DisplayName("Schema default values via $ref should not generate invalid Java")
+    void should_not_generate_invalid_java_for_schema_default_values() throws IOException {
+        Path path = findOpenAPIPath("issue-1216-schema-default.json");
+
+        OpenAPIToolsGenerator openAPIToolsGenerator = new OpenAPIToolsGenerator(
+                new QuarkusJavaServerCodegenConfigurator()
+                        .withInputBaseDir(path.toString())
+                        .withOutputDir(Files.createTempDirectory("").toString())
+                        .withBasePackage("org.issue1216schemadefault"));
+
+        List<File> files = openAPIToolsGenerator.generate();
+
+        List<File> modelFiles = files.stream()
+                .filter(file -> file.getName().endsWith("ConversionRequest.java"))
+                .toList();
+
+        assertThat(modelFiles).isNotEmpty();
+        for (File file : modelFiles) {
+            String content = Files.readString(file.toPath());
+            assertThat(content).doesNotContain("= {");
+        }
+    }
+
     private Path findOpenAPIPath(String specFileName) {
         URL url = this.getClass().getResource("/openapitools/" + specFileName);
         Objects.requireNonNull(url, "Could not find /openapi/" + specFileName);

--- a/server/deployment/src/test/resources/openapitools/issue-1216-object-default.json
+++ b/server/deployment/src/test/resources/openapitools/issue-1216-object-default.json
@@ -1,0 +1,83 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "API with object default values",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/v1/convert": {
+      "post": {
+        "summary": "Convert Document",
+        "operationId": "convertDocument",
+        "tags": ["Conversion"],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/ConversionRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "ConversionRequest": {
+        "type": "object",
+        "title": "ConversionRequest",
+        "properties": {
+          "options": {
+            "$ref": "#/components/schemas/ConvertDocumentsOptions"
+          }
+        }
+      },
+      "ConvertDocumentsOptions": {
+        "type": "object",
+        "title": "ConvertDocumentsOptions",
+        "description": "Options for document conversion.",
+        "properties": {
+          "from_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": ["docx", "pptx", "html", "image", "pdf", "asciidoc", "md"],
+            "description": "Input format(s) to convert from."
+          },
+          "to_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": ["md"],
+            "description": "Output format(s) to convert to."
+          },
+          "ocr_engine": {
+            "type": "object",
+            "title": "Ocr Engine Options",
+            "description": "OCR engine options.",
+            "default": {
+              "ocr_engine": "easyocr",
+              "lang": "en"
+            }
+          }
+        }
+      }
+    }
+  }
+}

--- a/server/deployment/src/test/resources/openapitools/issue-1216-schema-default.json
+++ b/server/deployment/src/test/resources/openapitools/issue-1216-schema-default.json
@@ -1,0 +1,166 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "API with object schema default value via $ref",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/v1/convert": {
+      "post": {
+        "summary": "Convert Document",
+        "operationId": "convertDocument",
+        "tags": ["Conversion"],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/ConversionRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "ConversionRequest": {
+        "type": "object",
+        "title": "ConversionRequest",
+        "properties": {
+          "options": {
+            "$ref": "#/components/schemas/ConvertDocumentsOptions"
+          }
+        }
+      },
+      "ConvertDocumentsOptions": {
+        "type": "object",
+        "title": "ConvertDocumentsOptions",
+        "description": "Options for document conversion.",
+        "default": {
+          "from_formats": ["docx", "pptx", "html", "image", "pdf", "asciidoc", "md"],
+          "to_formats": ["md"],
+          "image_export_mode": "embedded",
+          "do_ocr": true,
+          "force_ocr": false,
+          "ocr_engine": "easyocr",
+          "pdf_backend": "dlparse_v4",
+          "table_mode": "accurate",
+          "pipeline": "standard",
+          "document_timeout": 604800,
+          "abort_on_error": false,
+          "return_as_file": false,
+          "do_table_structure": true,
+          "include_images": true,
+          "images_scale": 2,
+          "do_code_enrichment": false,
+          "do_formula_enrichment": false,
+          "do_picture_classification": false,
+          "do_picture_description": false,
+          "picture_description_area_threshold": 0.05
+        },
+        "properties": {
+          "from_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "description": "Input format(s) to convert from."
+          },
+          "to_formats": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "description": "Output format(s) to convert to."
+          },
+          "image_export_mode": {
+            "type": "string",
+            "description": "Image export mode."
+          },
+          "do_ocr": {
+            "type": "boolean",
+            "description": "If enabled, the bitmap content will be processed using OCR."
+          },
+          "force_ocr": {
+            "type": "boolean",
+            "description": "If enabled, replace existing text with OCR-generated text."
+          },
+          "ocr_engine": {
+            "type": "string",
+            "description": "The OCR engine to use."
+          },
+          "pdf_backend": {
+            "type": "string",
+            "description": "The PDF backend to use."
+          },
+          "table_mode": {
+            "type": "string",
+            "description": "Mode to use for table structure."
+          },
+          "pipeline": {
+            "type": "string",
+            "description": "Choose the pipeline to process PDF or image files."
+          },
+          "document_timeout": {
+            "type": "integer",
+            "description": "The timeout for processing each document, in seconds."
+          },
+          "abort_on_error": {
+            "type": "boolean",
+            "description": "Abort on error if enabled."
+          },
+          "return_as_file": {
+            "type": "boolean",
+            "description": "Return the output as a zip file."
+          },
+          "do_table_structure": {
+            "type": "boolean",
+            "description": "If enabled, the table structure will be extracted."
+          },
+          "include_images": {
+            "type": "boolean",
+            "description": "If enabled, images will be extracted from the document."
+          },
+          "images_scale": {
+            "type": "number",
+            "description": "Scale factor for images."
+          },
+          "do_code_enrichment": {
+            "type": "boolean",
+            "description": "If enabled, perform OCR code enrichment."
+          },
+          "do_formula_enrichment": {
+            "type": "boolean",
+            "description": "If enabled, perform formula OCR, return LaTeX code."
+          },
+          "do_picture_classification": {
+            "type": "boolean",
+            "description": "If enabled, classify pictures in documents."
+          },
+          "do_picture_description": {
+            "type": "boolean",
+            "description": "If enabled, describe pictures in documents."
+          },
+          "picture_description_area_threshold": {
+            "type": "number",
+            "description": "Minimum percentage of the area for a picture to be processed."
+          }
+        }
+      }
+    }
+  }
+}

